auditor/perturbations/base.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod, abstractproperty
+from typing import List
+
+
+class AbstractPerturbation(ABC):
+    """Abstract class to aid in creation of perturbation classes
+    """
+    def __init__(self) -> None:
+        super().__init__()
+
+    @abstractmethod
+    def perturb(self) -> List[str]:
```

Comment:
shouldn't the base class perturb abstract method get the prompt as a parameter ?

Reply:
Great point! Added, although python does not enforce the arguments when the method is over-ridden.